### ISSUE TITLE
rust16 blockers

### ISSUE THREAD

**Issue by pghvlaans** [SlackBuildsOrg/slackbuilds#7526]
`rust16` is on hold due to a handful of FTBFS against `1.80.1` across all architectures. The cause appears to be an incompatibility with older versions of the `time` crate, which is currently at `0.3.36`. Affected scripts include:

- [x] `bottom` - `0.10.2` builds
- [ ] `lean-elan` - Fixed in `master` but not in latest release (x86_64 only)
- [x] `rustup` - `1.27.1` builds
- [ ] `wezterm` - Fixed in `main` but not in latest release

The current version of `rust16` is probably still months away from causing problems with newer stuff, so waiting on new versions and script updates is no problem for now. Just wanted to keep everyone in the loop.

**Comment by willysr** [comment]
@pghvlaans you can push PR and ask for WIP label so we won't merge it, but we can also test it using CI to make sure everything works just fine with new rust16 version

**Comment by willysr** [comment]
- [ ] ├ alacritty [dependees]  (Andrew Clemons <andrew.clemons@gmail.com>)
- [ ] ├ bat [dependees]  (Erich Ritz <erich.public@protonmail.com>)
- [ ] ├ bottom [dependees]  (K. Eugene Carlson <kvngncrlsn@gmail.com>)
- [ ] ├ cargo-c [dependees]  (Alexander Verbovetsky <alik@ejik.org>)
- [ ] ├ cargo-vendor-filterer [dependees]  (fourtysixandtwo <fourtysixandtwo@sliderr.net>)
- [ ] ├ clamav [dependees]  (Matteo Bernardini <ponce@slackbuilds.org>)
- [ ] ├ dust [dependees]  (Erich Ritz <erich.public@protonmail.com>)
- [ ] ├ fd [dependees]  (Andrew Clemons <andrew.clemons@gmail.com>)
- [ ] ├ greetd [dependees]  (Jay Lanagan <j@lngn.net>)
- [ ] ├ helvum [dependees]  (Lockywolf <for_sbo.helvum_2023-09-23@lockywolf.net>)
- [ ] ├ hyperfine [dependees]  (K. Eugene Carlson <kvngncrlsn@gmail.com>)
- [ ] ├ lean-elan [dependees]  (Lockywolf <for_sbo.lean-elan_2023-09-21@lockywolf.net>)
- [ ] ├ leftwm [dependees]  (Damian Perticone <mjolnirdam@gmail.com>)
- [ ] ├ mdbook [dependees]  (Pouria Rezaei <Pouria.rz@outlook.com>)
- [ ] ├ navi [dependees]  (Tonus <tonus1@free.fr>)
- [ ] ├ ncspot [dependees]  (K. Eugene Carlson <kvngncrlsn@gmail.com>)
- [ ] ├ newsboat
- [ ] ├ noip-duc
- [ ] ├ onefetch
- [ ] ├ python3-bcrypt
- [ ] ├ python3-maturin
- [ ] ├ ripgrep
- [ ] ├ rustup
- [ ] ├ skim
- [ ] ├ tzupdate
- [ ] ├ veles
- [ ] ├ wezterm
- [ ] └ yabridge

**Comment by willysr** [comment]
so far, those failed to builds are:

- [ ] development/lean-elan
- [ ] development/rustup

**Comment by willysr** [comment]
it's currently being tested here
https://github.com/SlackBuildsOrg/slackbuilds/pull/7561

**Comment by pghvlaans** [comment]
Everything so far is lining up with my own testing. `wezterm` should be the only other build fail.

**Comment by willysr** [comment]
development/lean-elan is still failing

**Comment by pghvlaans** [comment]
The outdated crate was replaced in [98ce78e](https://github.com/leanprover/elan/commit/97ce78e0e6aecdf3e8d35dbf42b0614302efb250) upstream, but there hasn't been a new release yet.

**Comment by willysr** [comment]
is it possible to cherry-pick some commits from master to fix? or use git commit version instead of the released one?

**Comment by pghvlaans** [comment]
It's sometimes possible to make a patch to use a different version of a particular crate (I did it with `ncspot` a few times to get it to build with 15.0's `rust`). I could see whether it works for `wezterm` and `lean-elan` if it would be better not to wait for new versions.

**Comment by pghvlaans** [comment]
Well, they both build with a crate replacement patch, but `wezterm` is basically unusable due to (I suppose) a text rendering issue. 

There's probably nothing for it but to either wait a month or two for new upstream versions or push to a git commit.

**Comment by willysr** [comment]
by that time, a new major version of rust16 will be released :)

**Comment by willysr** [comment]
@pghvlaans good news
https://github.com/leanprover/elan/issues/135

**Comment by pghvlaans** [comment]
Good news, indeed!

**Comment by willysr** [comment]
http://www.slackware.com/changelog/stable.php?cpu=x86_64
Fri Sep 6 19:22:57 UTC 2024
testing/packages/rust-1.81.0-x86_64-1_slack15.0.txz: Upgraded.

**Comment by pghvlaans** [comment]
Yeah, I tried building `wezterm` again today and it still errors out. 1.81.0 gives a helpful error message, though, so I suppose that's progress.

**Comment by pghvlaans** [comment]
Here's one potential solution. Would it be all right if I submitted a new script called `rust-opt` (probably with optional multilib capabilities) at version 1.81.0 and switched everything except for `wezterm` and `lean-elan` over to it? `rust16` could then be removed altogether once both of those upstream releases have occurred.

**Comment by willysr** [comment]
Sure thing

**Comment by pghvlaans** [comment]
Thanks. Once `rust-opt` has been approved, I'll make a PR with the dependency switches.
